### PR TITLE
feat(Tree): useVirtualTree hook + rewrite virtualizedTreePrototype using hook and react-window

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

- create `useVirtualTree` hook
- update existing `virtualizedTree` prototype with `useVirtualTree` hook and `react-window`

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/docs/package.json
Comment (layershifter) [inline packages/fluentui/docs/package.json:47]: Can be this dependency removed now?

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:49]:
```suggestion
```

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:44]: IMO, we can remove this as it's a custom component

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:90]: I suggest to add comment there that it will return `id` and this will make keys more stable instead of indexes

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:97]:
```suggestion
    listRef.current.resetAfterIndex(0);
```

Let's avoid `any` there

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:100]:
```suggestion
  const createTreeItem = React.useCallback<VirtualItemData["createTreeItem"]>(
```

Can we type this callback?

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:134]: Let's extract these two to real components, it also can be typed without type casts

```tsx
const OuterElement = React.forwardRef<HTMLDivElement>((props, ref) => <div ref={ref} {...props} role="none" />);
```

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:168]: Let's add a comment why this is need needed

File: packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts:21]: The easiest way is to use:

```
const listRef = React.useRef<VariableSizeList>();
```

However, it will put a dependency on `react-window` 💣 

```tsx
export interface VariableSizeListRef extends React.Component<any> {
  scrollTo(scrollOffset: number): void;
  scrollToItem(index: number, align?: 'auto' | 'smart' | 'center' | 'end' | 'start'): void;

  resetAfterIndex(index: number, shouldForceUpdate?: boolean): void;
}
```

```diff
-  listRef: React.MutableRefObject<any>;
+  listRef: React.MutableRefObject<VariableSizeListRef>;
```

```diff
-  const listRef = React.useRef();
+  const listRef = React.useRef<VariableSizeListRef>();
```

File: packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts:34]:
```suggestion
          listRef.current?.scrollToItem(focusIndex, 'center');
```

File: packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Tree/hooks/useVirtualTree.ts:87]:
```suggestion
        listRef.current?.scrollToItem(focusIndex, 'center');
```

**Review by layershifter — CHANGES_REQUESTED** [review]

File: packages/fluentui/docs/package.json
Comment (YuanboXue-Amber) [reply packages/fluentui/docs/package.json:47]: Nah.... it's in virtualized table prototype :D

**Review by layershifter — APPROVED** [review]
Please add a change entry 📜 and get a review from someone else before merge 👍

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (ling1726) [inline packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:135]: What happens if you omit the `width` prop ? is there a default set by `react-window` ?

**Review by ling1726 — APPROVED** [review]

File: packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx
Comment (YuanboXue-Amber) [reply packages/fluentui/docs/src/prototypes/VirtualizedTree/VirtualTree.tsx:136]: When I omit I get typing err because of @types/react-window :( 

There's no default from react-window. 
For list itself, the `width=-1` is passed to React Element is eventually ignored by browser. 
For items within the list, when the list is vertical, 'width=100%' is passed to React Element. 
(related part in react-window is here https://github.com/bvaughn/react-window/blob/d80bef25fe706d0c73fc801674c086f681811190/src/createListComponent.js#L343)

**Review by miroslavstastny — APPROVED** [review]
LGTM.
But I am concerned about 0 unit tests we have for `useTree` and `useVirtualTree`. We should definitely cover the hooks by UTs.

**Pull request merged**
